File: src/Problem/H_index.java
class Solution {
    public int hIndex(int[] citations) {
        // 개수와 크기로 나눈다.
        // 크기 <= 개수 인 최대 크기를 찾아야한다.
        
        int num = 0; // 개수
        int citationsLength = citations.length;
        int citation = 0; // 크기 : citations중 최대의 숫자로 디폴트 값 정함.
        
        // citations중 최대의 숫자 알아내기
        for (int i = 0; i < citationsLength; i++){
            if(citation < citations[i]){
                citation = citations[i];
            }
        }
        
        // 크기 <= 개수 알아내기.
        for(; citation >= 0; citation--){
            num = 0;
            for (int i = 0; i < citationsLength; i++){
                if(citation <= citations[i]){
                    num++;
                }
            }
            if(citation <= num){
                break;
            }
        }
        return citation;
    }
}

File: Leetcode/Problem/H_index.py
class Solution:
    def hIndex(self, citations: List[int]) -> int:
        citations.sort()
        count=0
        citations_length = len(citations)
        output = citations_length

        if citations_length is 0:
            return 0

        if citations_length is 1:
            if citations[0] is 0:
                return 0
            return 1

        for i in range(citations_length):
            for j in range(citations_length):
                if citations[j]>=output:
                    count+=1

            if count >= output:
                break
            output -= 1
            count=0

        return output
